src/coreclr/src/System.Private.CoreLib/src/System/StartupHookProvider.cs

```diff
@@ -24,6 +30,9 @@ private struct StartupHookNameOrPath
         // containing a startup hook, and call each hook in turn.
         private static void ProcessStartupHooks()
         {
+            if (!IsSupported)
+                return;
```

Comment:
I would structure this so that if someone specifies a startup hook (there's something actionable in STARTUP_HOOKS) and startup hook support was linked out, they get some sort of feedback about it (exception?).

Reply:
I think it is better to ignore it silently (no exception).

Having ability to disable startup hooks per app is useful on its own, independent of linking. For example, STARTUP_HOOKS can be set globally for APM, but there is one utility where you do not want it to kick in.